### DISCUSSION TITLE
How to define a Spot Color/CutContour ?

### DISCUSSION THREAD

**Discussion by silikone** [prawnpdf/prawn#1201] [Q&A]
Hi,

I'm using Prawn to generate a lot of automatic print materials.
I need to define a 'CutCountour' to indicate where the printers are to cut the materials.

For a quick brush up on CutCountour check this:
https://www.prigraphics.com/wp-content/uploads/2014/04/How-to-Prepare-Contour-Cut-Jobs.pdf

As far as I see this, it is a named color(Spot Color) that is used to draw a shape to indicate where the cutting is to happen.
Ideally placed on a separate layer from the rest of the graphics/text.

This was a very useful discussion - but seemed to end unresolved
https://groups.google.com/g/prawn-ruby/c/L-_6rcX6OhQ/m/IDE6QnQh62EJ

Is there a way to create this in Prawn??

And if not - any other options that I should look into?
 
/S

**Comment by pointlessone** [comment]
I personally is not familiar with the concept. If I read the linked document correctly the idea is to embed a special path that will be used as a cut contour.

If I read the instructions correctly, you need to define a path for the cut out. A cursory examination seem to indicate that it makes use of tagged content in the PDF. At the moment Prawn doesn't support it natively.

It is possible to implement both marked content and addition of all the supporting objects but it's a rather involved and not documented process. 

You may draw some inspiration from [Images](https://github.com/prawnpdf/prawn/blob/master/lib/prawn/images.rb) module. It's fairly self-contained and demonstrates both objects and content stream additions.

**Comment by afdev82** [comment]
I have also a similar question, the printing company requires me to draw some shapes on another "channel" using a spot color in order to overprint it on other graphics printed before with the normal inks. In this way they can specify another ink to use for this extra channel.
I have a tutorial explaining how to add it using Photoshop, I'm interested in the a) or c) techniques as listed in the pdf attached.
I tried the first method in Photoshop and as a result in Acrobat I can see the extra layer as a totally different "color" (see the attached image).
The only way I found in the docs to add an extra channel (alpha channel) is using a soft mask. I was trying that to see the result, but I had some problems, anyway I'm not sure I could use that as a workaround.

[GrandWhite.pdf](https://github.com/prawnpdf/prawn/files/7066026/GrandWhite.pdf)
[test.pdf](https://github.com/prawnpdf/prawn/files/7066033/test.pdf)

<img width="1440" alt="Screen6" src="https://user-images.githubusercontent.com/4134610/131110601-ed07fab2-5b11-4e1e-a5c3-06ebcc4b837d.png">

Is it possible to read the test file in some way to identify what is needed to implement this feature or if a workaround is possible to achieve that result?
Thank you for your support!

**Reply by gettalong** [reply]
It seems that in the content stream of the test PDF marked content operators are used to define things like a "MainChannel", "MultiChannelImage" and "MainChannelImage". Alas, I don't know how that translate to what you are seeing ...

This may also just be an artifact of how PhotoShop does things and that the really only used part is the Separation color space...

**Comment by emre-basala** [comment]
Any updates on this? I am looking into defining Spot Colors in the PDF as well. Thanks in advance!

**Comment by ChrisFreemanKY** [comment]
I don't think there's a way to do this in Prawn but there's a web app that does it at cutcontour.com that may help.  Itextsharp library also supports spot color I've used it before.
